### PR TITLE
echo.py: a simple plugin for testing purposes

### PR DESCRIPTION
**Pull request by dpino**
This is a simple plugin to showcase how a matrix plugin works.

To run it, simply go to `examples/echo-test` directory and run:

```bash
$ ./echo-test --user <user> --pasword <password>
```

There's one thing that I couldn't get working though and it's that the `async` function doesn't get called periodically, so in order to print out a message in a matrix room I had to call the method that prints out a message explicitly in the constructor.

### CODE REVIEW

File: examples/echo-test/echo-test [hunk 1]
```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env bash
+
```
Comment (psaavedra) [inline examples/echo-test/echo-test:2]: Why not a `matrix-echo` bin under the tool directory and add it in the installable scripts list (https://github.com/psaavedra/matrix-bot/blob/master/setup.py#L51) .

File: examples/echo-test/echo-test-template.cfg [hunk 2]
```diff
@@ -0,0 +1,26 @@
+settings['DEFAULT'] = {
```
Comment (psaavedra) [inline examples/echo-test/echo-test-template.cfg:1]: The same, if you moves the echo-test to tools/, you could also move the echo-test-template.cfg as a `matrix-echo.cfg.example` like https://github.com/psaavedra/matrix-bot/blob/master/cfg/matrix-bot.cfg.example and add it in the setup.py file https://github.com/psaavedra/matrix-bot/blob/master/setup.py#L47

**Review by psaavedra — CHANGES_REQUESTED** [review]

File: setup.py [hunk 3]
```diff
@@ -44,11 +44,12 @@ def exec_file(path_segments):
     url='http://github.com/psaavedra/matrix-bot',
     packages=find_packages(),
     package_data={
-        "matrixbot": [ "../cfg/matrix-bot.cfg.example" ]
+        "matrixbot": [ "../cfg/matrix-bot.cfg.example", "../cfg/echo-test-template.cfg" ]
```
Comment (psaavedra) [inline setup.py:47]: break the line in two like below in the `scripts` list definition.

**Review by psaavedra — CHANGES_REQUESTED** [review]
Commented inline.

File: tools/matrix-echo [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+trap cleanup SIGINT SIGHUP EXIT
+
+scan_args
+
+# Try to fetch credentials from .git-credentials.
+if [[ -f "$HOME/.git-credentials" ]]; then
+   line=$(grep "gitlab.igalia.com" ~/.git-credentials)
+   if [[ "$?" -eq "0" ]]; then
+      credentials=$(echo "$line" | sed -r 's|https?://(\w+):(\w+)@gitlab.igalia.com|\1:\2|g')
```
Comment (psaavedra) [inline tools/matrix-echo:27]: Could be this domain an parameter like the credentials are?

**Comment by psaavedra** [comment]
> There's one thing that I couldn't get working though and it's that the `async` function doesn't get called periodically, so in order to print out a message in a matrix room I had to call the method that prints out a message explicitly in the constructor.

The bot attempts to reply in an already created 1-on-1 chat with the user. I think that what is happening to you is that the replied is being sent to you in a different room.

File: tools/matrix-echo (hunk 4 above)
Comment (dpino) [reply tools/matrix-echo]: Yep, I can search for the 'domain' in the template config.

**Comment by dpino** [comment]
In the end I refactored the whole script and template files to remove any reference any specific settings. The bot can now be started with the default template and the following arguments:

```bash
$ tools/matrix-echo --uri https://example.com --room "#matrix-room"
```

File: tools/README.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    'only_local_domain': False
+}
+
+echo_plugin = {
+    'module': "matrixbot.plugins.echo",
+    'class': "EchoPlugin",
+    'settings': {
+        'message': 'hello room!',
+        'username': 'USER',
+        'rooms': ['!xWYVwSiiyuwZkEEGEQ:igalia.com']
```
Comment (psaavedra) [inline tools/README.md:30]: Can we use a dummy reference here instead of :igalia.com?

**Comment by psaavedra** [comment]
Other than that LGTM.

**Pull request merged**